jerry-core/parser/js/js-parser-statm.c

```diff
@@ -804,9 +804,9 @@ parser_parse_function_statement (parser_context_t *context_p) /**< context */
   literal_p->u.bytecode_p = compiled_code_p;
   literal_p->type = LEXER_FUNCTION_LITERAL;
 
-#if ENABLED (JERRY_ESNEXT)
+#if ENABLED (JERRY_MODULE_SYSTEM)
   parser_compiled_code_set_function_name (context_p, compiled_code_p, function_name_index, 0);
-#endif /* ENABLED (JERRY_ESNEXT) */
+#endif /* ENABLED (JERRY_MODULE_SYSTEM) */
```

Comment:
Are you sure this should be module guarded? This seems more related to function names than modules.

Reply:
True, with this the function will not have a name in case of disabled module system. Will update.

Reply:
@dbatyai I've update the PR.